test/e2e/e2e_test.go

```diff
@@ -92,6 +106,15 @@ func TestE2E(t *testing.T) {
 	}
 
 	gomega.RegisterFailHandler(ginkgo.Fail)
+
+	// Ensure all pods are running and ready before starting tests (otherwise,
+	// cluster infrastructure pods that are being pulled or started can block
+	// test pods from running, and tests that ensure all pods are running and
+	// ready will fail).
+	if err := waitForPodsRunningReady(api.NamespaceDefault, minStartupPods, podStartupTimeout); err != nil {
```

Comment:
I find this rather confusing.  We say we're waiting for all pods to be ready, but in the current configuration we only wait for any one pod to be ready.  Also, should individual tests that require specific pods to be running in order to succeed not perform their own specific checks against the states of those pods.

Otherwise stated, I don't understand how ensuring that one pod is running in the default namespace is good enough to be useful to any tests on a reliable basis?

I'm coming in rather late here, so it's quite possible that I'm missing some context here, in which case please accept my apologies in advance.